suite-common/walletconnect/src/adapters/ethereum.ts

```diff
@@ -0,0 +1,164 @@
+import { WalletKitTypes } from '@reown/walletkit';
+
+import { createThunk } from '@suite-common/redux-utils';
+import { getNetwork } from '@suite-common/wallet-config';
+import { selectAccounts, selectSelectedDevice } from '@suite-common/wallet-core';
+import { getAccountIdentity } from '@suite-common/wallet-utils';
+import TrezorConnect from '@trezor/connect';
+import * as trezorConnectPopupActions from '@trezor/suite-desktop-connect-popup';
+
+import { WALLETCONNECT_MODULE } from '../walletConnectConstants';
+import { WalletConnectAdapter } from '../walletConnectTypes';
+
+const ethereumRequestThunk = createThunk<
+    void,
+    {
+        event: WalletKitTypes.SessionRequest;
+    }
+>(`${WALLETCONNECT_MODULE}/ethereumRequest`, async ({ event }, { dispatch, getState }) => {
+    const device = selectSelectedDevice(getState());
+    const getAccount = (address: string, chainId?: number) => {
+        const account = selectAccounts(getState()).find(
+            a =>
+                a.descriptor.toLowerCase() === address.toLowerCase() &&
+                a.networkType === 'ethereum' &&
+                (!chainId || getNetwork(a.symbol).chainId === chainId),
+        );
+        if (!account) {
+            throw new Error('Account not found');
+        }
+
+        return account;
+    };
+
+    switch (event.params.request.method) {
+        case 'personal_sign': {
+            const [message, address] = event.params.request.params;
+            const account = getAccount(address);
+            const response = await dispatch(
+                trezorConnectPopupActions.connectPopupCallThunk({
+                    id: 0,
+                    method: 'ethereumSignMessage',
+                    payload: {
+                        path: account.path,
+                        message,
+                        hex: true,
+                        device,
+                        useEmptyPassphrase: device?.useEmptyPassphrase,
```

Comment:
totally ignorable, just a thought that maybe path, device, and useEmptyPasshrase could be pull out to some common param. feel free to ignore

Reply:
Device probably shouldn't even be needed here, since it will get overridden inside connectPopupCallThunk